packages/engine-server/src/markdown/remark/utils.ts

```diff
@@ -216,6 +219,47 @@ export class LinkUtils {
   }
 }
 
+export class AnchorUtils {
+  static async findAnchors(opts: {
+    note: NoteProps;
+    wsRoot: string;
+  }): Promise<{ [index: string]: DNoteAnchorPositioned }> {
+    if (opts.note.stub) return {};
+    const noteContents = await NoteUtils.readFullNote(opts);
+    const noteAnchors = RemarkUtils.findAnchors(noteContents);
+    const anchors: [string, DNoteAnchorPositioned][] = [];
+    noteAnchors.forEach((anchor) => {
+      if (_.isUndefined(anchor.position)) return;
+      const { line, column } = anchor.position.start;
+      if (anchor.type === "heading") {
+        const value = getSlugger().slug(anchor.children[0].value as string);
+        anchors.push([
+          value,
+          {
+            type: "header",
+            value,
+            line: line - 1,
+            column: column - 1,
+          },
+        ]);
+      } else if (anchor.type === DendronASTTypes.BLOCK_ANCHOR) {
+        anchors.push([
+          `^${anchor.id}`,
+          {
+            type: "block",
```

Comment:
we should be able to use the value of `DendronASTTypes.BLOCK_ANCHOR` here - makes it easier to find and refactor

Reply:
The values for the AST types and anchor types are different though, `blockAnchor` and `heading` vs. `block` and `header`. Should I change them to match the AST type names? Or I can create an enum for anchor types if we just want to avoid magic numbers.

Reply:
unless there's a good reason not to, I would just have the names be the same. its already the case for `wikiLink` and the other dendron constructs